### PR TITLE
🔧 Devops: Switch to `ruff` for linting/formatting

### PR DESCRIPTION
**Pull request by mbercx**

### CODE REVIEW

**Review by t-reents — DISMISSED** [review]
LGTM!

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,50 +1,30 @@
...
     hooks:
-    -   id: double-quote-string-fixer
     -   id: end-of-file-fixer
     -   id: fix-encoding-pragma
     -   id: mixed-line-ending
     -   id: trailing-whitespace
 
--   repo: https://github.com/ikamensh/flynt/
-    rev: '0.76'
+-   repo: https://github.com/astral-sh/ruff-pre-commit
```
Comment (sphuber) [inline .pre-commit-config.yaml:10]: `ruff` does import ordering as well, so you can remove isort as well now

File: pyproject.toml [hunk 2]
```diff
@@ -34,7 +34,7 @@ Source = 'https://github.com/aiidateam/qe-tools'
 [project.optional-dependencies]
 dev = [
     'codecov',
-    'mypy==0.930',
+    'mypy~=1.11',
     'pre-commit',
     'timeout-decorator',
     'pylint~=2.16.0',
```
Comment (sphuber) [inline pyproject.toml:40]: Remove `pylint` requirement

**Review by sphuber — CHANGES_REQUESTED** [review]

**Comment by mbercx** [comment]
@sphuber do you still want to have a look at this one? Or do I go ahead?

**Comment by sphuber** [comment]
> @sphuber do you still want to have a look at this one? Or do I go ahead?

All good, please go ahead. No need to check with me

**Pull request merged**